powermax/provider/volume_data_source.go

```diff
@@ -531,23 +533,42 @@ func (d *volumeDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 }
 
 // updateVolumeState iterates over the volume list and update the state.
-func updateVolumeState(ctx context.Context, p *client.Client, params map[string]string) (response []models.VolumeDatasourceEntity, err error) {
-	volIDs, err := p.PmaxClient.GetVolumeIDListWithParams(ctx, p.SymmetrixID, params)
+func updateVolumeState(ctx context.Context, p *client.Client, params powermax.ApiListVolumesRequest) (response []models.VolumeDatasourceEntity, err error) {
+	volIDs, _, err := params.Execute()
 	if err != nil {
-		return nil, err
-	}
-	for _, volumeID := range volIDs {
-		volResponse, err := p.PmaxClient.GetVolumeByID(context.Background(), p.SymmetrixID, volumeID)
-		if err != nil {
+		err1, ok := err.(*powermax.GenericOpenAPIError)
```

Comment:
@doriac11 you can think of using helper.GetErrorString

Reply:
I will refactor to use the method, I didnt know about it until creating the PR.